containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -382,6 +382,12 @@ class UnorderedMap {
     static_assert(
         !alloc_prop_t::has_pointer,
         "Allocation properties should not contain the 'pointer' property.");
+    static_assert(
+        alloc_prop_t::sequential_host_init &&
+            (SpaceAccessibility<
+                HostSpace, typename execution_space::memory_space>::accessible),
+        "SequentialHostInit only allowed if the UnorderedMap's MemorySpace is "
+        "accesible to the host.");
```

Comment:
We already have a `static_assert` at https://github.com/kokkos/kokkos/blob/28848dfa8fc1411873b6a663c4d29ba05712510e/core/src/View/Kokkos_ViewAlloc.hpp#L200.